src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -1376,19 +1374,7 @@ PyMODINIT_FUNC PyInit_lpython_module_)" + fn_name + R"((void) {
         if( is_target_list && is_value_list ) {
             ASR::List_t* list_target = ASR::down_cast<ASR::List_t>(ASRUtils::expr_type(x.m_target));
             std::string list_dc_func = c_ds_api->get_list_deepcopy_func(list_target);
-            if (ASR::is_a<ASR::Var_t>(*x.m_target)) {
-                ASR::symbol_t *target_sym = ASR::down_cast<ASR::Var_t>(x.m_target)->m_v;
-                if (ASR::is_a<ASR::Variable_t>(*target_sym)) {
-                    ASR::Variable_t *v = ASR::down_cast<ASR::Variable_t>(target_sym);
-                    if (v->m_intent == ASRUtils::intent_out) {
-                        src += indent + list_dc_func + "(&" + value + ", " + target + ");\n\n";
-                    } else {
-                        src += indent + list_dc_func + "(&" + value + ", &" + target + ");\n\n";
-                    }
-                }
-            } else {
-                src += indent + list_dc_func + "(&" + value + ", &" + target + ");\n\n";
-            }
```

Comment:
Should be reverted as per https://github.com/lcompilers/lpython/commit/f9ab31ebc99f6a76f9d3d1e162e37e667e550d05